test/NameBinding/dynamic-member-lookup.swift

```diff
@@ -0,0 +1,231 @@
+// RUN: %target-swift-frontend -typecheck -verify %s
+var global = 42
+
+@dynamicMemberLookup
+struct Gettable {
```

Comment:
It's important to test `subscript(dynamicMember:)` within generic types and probably as a generic subscript.

Reply:
👍 Thank you for the test cases, added.